core/plannerShared.cpp

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "plannerShared.h"
+#include "core/pref.h"
+#include "core/settings/qPrefDivePlanner.h"
+
+
+plannerShared *plannerShared::instance()
+{
+    static plannerShared *self = new plannerShared;
+    return self;
+}
+
+// Used to convert between meter/feet and keep the qPref variables independent
+#define UNIT_FACTOR ((prefs.units.length == units::METERS) ? 1000.0 / 60.0 : feet_to_mm(1.0) / 60.0)
```

Comment:
Maybe use a name that signals this is a velocity unit and that it is internal velocity / user velocity (so it is obvious if you have to multiply or divide).

Reply:
How about <variable name>Unit ? 

I totally agree that different names would be best, but I failed to come up with a convincing name.

Reply:
what about TO_MM_BY_SEC ?

Reply:
Just to be sure, I do not misunderstand you.

You talk ONLY about the macro, not the variable names (ascratelast6m etc.) which are exposed externally to plannerShared ?

If you only talk about the macro, I will make the change immediately.

Reply:
Yea, this UNIT_FACTOR convertion thingie I didn't like at all. I know It's there since before, but I think we should clean that up.

We have a velocity_t in core/profile.h , but thats rather just colors for the profile. I think we should have a propper velocity_t which describes this and convertion helpers around that.

I think mm/sec is the right unit for a velocity_t for the planner.

Reply:
I have put a "velocity_t" on my agenda, but I would prefer to have this simple step first, because it is very easy to review, that I have not broken anything, and then in a second step replace the macro with a proper velocity_t.

Reply:
I'm completely fine with that.